Sources/MapboxMaps/Foundation/Camera/CameraManager.swift

```diff
@@ -174,122 +173,109 @@ public class CameraManager {
     ///   - duration: If animated, how long the animation takes
     ///   - animation: closure to perform
     ///   - completion: animation block called on completion
-    fileprivate func performCameraAnimation(duration: TimeInterval, animation: @escaping () -> Void, completion: ((UIViewAnimatingPosition) -> Void)? = nil) {
+    fileprivate func performCameraAnimation(duration: TimeInterval, animation: @escaping (inout CameraTransition) -> Void, completion: ((UIViewAnimatingPosition) -> Void)? = nil) {
 
         // Stop previously running animations
-        internalCameraAnimator?.stopAnimation()
+        internalAnimator?.stopAnimation()
 
         // Make a new camera animator for the new properties
-        internalCameraAnimator = makeCameraAnimator(duration: duration,
-                                      curve: .easeOut,
-                                      animationOwner: .custom(id: "com.mapbox.maps.cameraManager"),
-                                      animations: animation)
+
+        let cameraAnimator = makeAnimator(duration: duration,
+                                                curve: .easeOut,
+                                                animationOwner: .custom(id: "com.mapbox.maps.cameraManager"),
+                                                animations: animation)
 
         // Add completion
-        internalCameraAnimator?.addCompletion({ [weak self] (position) in
+        cameraAnimator.addCompletion({ [weak self] (position) in
             completion?(position)
-            self?.internalCameraAnimator = nil
+            self?.internalAnimator = nil
```

Comment:
what if it was completed due to being interrupted by a different one?

Reply:
This has been reordered so shouldn't be an issue anymore.